packages/template-retail-react-app/app/hooks/use-auth-modal.js

```diff
@@ -38,69 +44,92 @@ export const AuthModal = ({
     onLoginSuccess = noop,
     onRegistrationSuccess = noop,
     onPasswordResetSuccess = noop,
+    isOpen,
+    onOpen,
+    onClose,
     ...props
 }) => {
     const {formatMessage} = useIntl()
-    const customer = useCustomer()
+    const customerId = useCustomerId()
+    const customerType = useCustomerType()
+    const customer = useCustomer(
+        {customerId},
+        {enabled: !!customerId && customerType === 'registered'}
+    )
+    const isRegistered = customerType === 'registered'
     const navigate = useNavigation()
     const [currentView, setCurrentView] = useState(initialView)
     const form = useForm()
     const submittedEmail = useRef()
     const toast = useToast()
+    const login = useShopperLoginHelper(ShopperLoginHelpers.LoginRegisteredUserB2C)
+    const register = useShopperLoginHelper(ShopperLoginHelpers.Register)
 
     const submitForm = async (data) => {
         form.clearErrors()
 
-        return {
-            login: handleLogin,
-            register: handleRegister,
-            password: handleResetPassword
-        }[currentView](data)
-    }
+        // mutation is cached by default?
+        // TODO: should we disable mutation cache by default?
+        login.reset()
+        register.reset()
```

Comment:
I'm running into an issue that it seems mutations results are cached? Having `reset` all over the places seems a bad developer experience. Looking into a way to disable mutation cache by default.

Reply:
🤔 Are you saying that by default if a customer was to login successfully, any login attempts mad after with the same username and password would also be automatically be returned as successes? Even if the users account was disabled after the first login?

If that is true, I think that is not a reasonable default. You could change the default mutation options to have a cacheTime of 0 as a solution.